### PR TITLE
Cat/Xeno Teeth Coins

### PR DESCRIPTION
**Pull request by ghost**
You can weld together 32 cat teeth/48 Xeno teeth to make a special coin that gives really good shit (from vendors) for greytiders and security alike. Encourages people to hunt cats.

Current vending machine items are:

insulated gloves(5), retro lasers(2), switchblades(2), telescopic batons(2), foam riots darts(5) (for YouTools)

and advanced tasers, lasers, disablers(2 all) for security vendors.

Cat/Xeno teeth can be welded together by clicking a full set with a welding tool. Suggestions/head input appreciated.

### CODE REVIEW

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -803,6 +861,8 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 					/obj/item/weapon/reagent_containers/food/snacks/donut = 14,/obj/item/weapon/storage/box/evidence = 6,/obj/item/device/flashlight/seclite = 6)
 	contraband = list(/obj/item/clothing/glasses/sunglasses = 2,/obj/item/weapon/storage/fancy/donut_box = 2)
 	premium = list(/obj/item/weapon/coin/antagtoken = 1)
+	special = list(/obj/item/weapon/gun/energy/laser/captain/scattershot = 10, /obj/item/weapon/claymore = 10, /obj/item/weapon/katana = 10, /obj/item/weapon/gun/energy/laser/captain = 10)
```
Comment (danhibiki) [inline code/game/machinery/vending.dm]: Fuck no

Comment (TDcoolguy300) [inline code/game/machinery/vending.dm]: Yeah i am with Dan here, having the Antag objectives in a vending machine is not a good thing. Plus that fucking Katana/claymore is stupidly OP

Comment (danhibiki) [inline code/game/machinery/vending.dm]: These are the bonus equipment were giving security at that. Like we don't have a problem with them using lethal force enough, wait until we tell them you can farm for a katana with enough catteeth.

**Comment by TDcoolguy300** [comment]
I dont mind the overall concept, but execution could of been done better with the vending machine contents.

File: code/game/machinery/vending.dm (hunk 1 above)
Comment (ghost) [inline code/game/machinery/vending.dm]: What would be a better replacement weapon? It's pretty hard to get to the 1-3 or so cats on the station before the greytide, it should pay off to collect a whole set of cat teeth

**Comment by ghost** [comment]
@TDcoolguy300 Any suggestions for contents, then? Cats aren't that easy to find with a full set of teeth intact, the vermin are getting too good at hiding, so I was hoping for something better than what's in the armory.

**Comment by danhibiki** [comment]
I think thats your problem, you are setting the standard for this "prize" entirely too high for the effort.

**Comment by ghost** [comment]
What do you suggest, then? It seems a bit pointless to give vendors something that the user could get without much trouble on their own.

**Comment by danhibiki** [comment]
If I was going to go off the walls, I would try to go with something with a bit less power creep in the current station. Its easy to find a highbrute/burn damage weapon, but things like stamina damage is low, why not toss in a box of riot foam darts?

**Comment by danhibiki** [comment]
Instead of going for items that will just result in loldeath for people who get dunked by it, look for more utility that people might be interested in that doesn't directly involve dunking someone.

**Comment by ghost** [comment]
Do those need a gun to fire or what? And disablers could do stamina damage as well.

**Comment by danhibiki** [comment]
the toy guns are easy to find, you can actually get three riot darts as a nontraitor right now. The darts and guns are located in the toy gun arena which is north of chapel thru maintenance, but i'm fairly certain you can win a toy gun thru arcade as well.

**Comment by ghost** [comment]
Alright, I'll take a look at those and add them for the engivends, and maybe disablers. Telescopic spears are also fun. I'm considering adding fire axes, or are those too OP/job-related?

**Comment by Nokington** [comment]
Yeah no antag weapons.

**Comment by ghost** [comment]
@Nokington Alright then. I'll PR some new lists in a minute, tell me what you think about it.

In hindsight adding really robust shit to engivends is also a bad idea because fun hating antags might just blow it up

**Comment by ghost** [comment]
Updated. The YouTool now gives all-access, fire axes, and disablers(in addition to previous items), whereas the SecVend gives less powerful shit.

**Comment by Spacedong** [comment]
>all access

you've gotta be fucking kidding me 

>disablers

sec already has these, they're called hybrid tasers

>fireaxe

there's already two on the map, how many do you want

**Comment by ghost** [comment]
@Spacedong Not sure if you're complaining that these are too powerful or too weak

**Comment by Spacedong** [comment]
@cherkir they're dumb, all access is a horrible prize, disablers is just dumb, fire axes are OP and there's already 2 on station

**Comment by ghost** [comment]
@spacedong What would you suggest? And the vendor also gives switchblades/retro lasers/gloves

**Comment by Pyko1** [comment]
I'd go with telebatons, riotfoam darts, and beartraps at top.

**Comment by ghost** [comment]
> beartraps

uhhh

**Comment by Pyko1** [comment]
of course you need traps if you are going to became a master cat hunter.

**Comment by ghost** [comment]
Yeah, but is 32 cat teeth worth a measly bear trap?

**Comment by Pyko1** [comment]
isn't that like 1 cat, who can't even defend himself, sure might give a box of 2 or 3 for it if feeling generous.

**Comment by ghost** [comment]
Yeah, but unless less round was a griffest there usually aren't that many cats, whereas this PR will cause super high demand for cat teeth

Riot darts do stamina damage, right? If so, they can replace disablers. And telebatons or telespears are dank.

**Comment by ghost** [comment]
Also apparently riot darts are pretty robust, 25 damage each. Is damage equal to force for those, or what?

**Comment by Pyko1** [comment]
Telebatons, since if you want a telespear you can easily craft it from that.

Usually I have no issues getting 2 or 3 cats when I go cat hunting, and that's not even counting cloning the cats just to kill em again.

I actually once had 2 tanks full of cat blood, along with multiple full cat teeth necklaces.

**Comment by ghost** [comment]
Holy shit

I'll replace disablers and fire axes with telebatons and riotfoam darts then

How does the new shitcurity list look?

**Comment by Spacedong** [comment]
this just sounds like a dumb idea in all honesty tbh fam

**Comment by ghost** [comment]
@spacedong what are you, a cat sympathizer

**Comment by Pyko1** [comment]
/obj/item/weapon/pneumatic_cannon 
You do realize this is pretty much deadliest thing on the station, even with embed nerf.
You can load it with 10 kitchen knifes, 10 throwing starts, 20 syndie cards or 20 shards. Even the ghetto version is still really strong.

/obj/item/weapon/grenade/clusterbuster
And these leave pretty much everyone blind and deaf.

/obj/item/weapon/gun/energy/ionrifle
And these guns are limited for a reason, they fuck up malf AI, and borgs so hard, hell you don't even need to score a direct hit.

/obj/item/weapon/card/id/captains_spare
NOPE!

Also you do realize that genetists can make endless amount of cats from monkeys if they get dna  and changelings can just use trasform stign to make make cats.

**Comment by ghost** [comment]
@Pyko1 Yeah, I actually realized that a while ago, the problem is that if they don't do that there are a very limited number of cats. Do teeth change when someone gets turned into a cat? Because they might load on roundstart, which would make that irrelevant

Not sure what's wrong with the spare ID, and the ionrifle is overpowered on a round that only happens 10% of the time. If the AI is discreet nobody has a reason to purchase them.

I can remove clustergrenades and replace the cannon with a hybrid taser. And I could fix the too many cats problem by limiting the amounts of things in vending machines.

**Comment by Spacedong** [comment]
why would you need a hybrid taser when the armory has 20 of them, spare ID is just horrible, it can be used to GIVE EVERYONE ALL ACCESS and lead to lynching of HOP, it's also a high risk item as well.

ion rifles can fuck up a lot more than borgs and AI, they can fuck over mechs, nuke ops, etc 

please stop embarassing yourself lad just close this

**Comment by Pyko1** [comment]
Well if this were to went live, the number 1 thing for CMO, genetist and lings would be to secure a cat, and it would be even safe since its valid salid. Then they just get the genes, and enjoy more guns than they can carry, along with all access and insulated. 

Also there is no need of having 10 of all items, that just makes it even worthier to create cats for teeth farming. Should be 1 or 2 at tops, so better get those teeths fast. Also the items  should not be anything game changing really, just fancy/funny stuff that might or might not be wanted by loitering assistants not being busy by antagging the shit up.

**Comment by ghost** [comment]
Updated PR that includes a nerf of both sec/assistant shit and limits quantity of items.

However I'm almost positive that teeth do not change with species, as they are created in carbon/human/New()

**Comment by Pyko1** [comment]
when you change monkey to human, I recall it creates carbon/human/New()

**Comment by Pyko1** [comment]
since monkeys are not human

**Comment by Spacedong** [comment]
just remove every weapon/tool and add cool costumes

**Comment by ghost** [comment]
I didn't consider changing monkeys, just injecting genes into humans. Anyway, it doesn't matter now since most of the powerful items are limited to a quantity of 2.

@spacedong what costumes

**Comment by doobiedoo23** [comment]
Good concept, shit execution

**Comment by ghost** [comment]
@Pyko1 Lings don't regenerate teeth.

@Doobiedoo23 Did you check out the newest commit? If so, what do

**Comment by Spacedong** [comment]
just close this

**Comment by Spacedong** [comment]
thats what u do

**Comment by ghost** [comment]
F I G H T  M E @Spacedong

**Comment by Spacedong** [comment]
look lad, this is terrible, either add items of no value or don't add anything at all

**Comment by Pyko1** [comment]
well monkey form and back to cat will restore em most likely, since i really doubt anyone anyone made a check for that since it was not an issue before.

Quite frankly, if you want to add any meaningful loot, any way of creating more cat teeth has to be patched first.

**Comment by ghost** [comment]
@Spacedong Why would anyone spend half their round hunting a cat to get an item of no value

@Pyko1 All of the useful stuff is only 2 in quantity now, but I can check if that demonkeying actually does anything.

**Comment by Spacedong** [comment]
except cats are easy to dunk if you're good

**Comment by Pyko1** [comment]
I often spend whole round's hunting cats, for no other reward than the thrill of the hunt. There is no need really for anything besides pat on a back for killing a pathetic creature that cant even defend itself.

**Comment by ghost** [comment]
But that is the purpose of this PR. People are starting to sympathize with cats because they have no motive to kill them.

**Comment by Spacedong** [comment]
except there is a motive, gas furfags

**Comment by ghost** [comment]
I have been on multiple cat rounds where Security/heads offered to protect me despite the fact that I've  played very few

**Comment by Pyko1** [comment]
There is always people willing to dunk cats, and cat lovers. If anything cat lovers should be made valid as well, if they are actively trying to help em.

**Comment by Spacedong** [comment]
CLOSE THIS

**Comment by ghost** [comment]
@Pyko1 True. By the way, from the looks of things going from monkey to human doesn't do anything to external organs, IE, teeth.

So cat teeth are not regenerateable.

**Comment by Spacedong** [comment]
you are literally filling about 5 screens worth of logs in coderchat with comments on this PR

**Comment by ghost** [comment]
You have made 40% of the last ten comments

**Comment by Spacedong** [comment]
you've made 23 comments while i've made like 10

**Comment by ghost** [comment]
This PR was opened 12 hours ago

You started commenting 1 hour ago

You have the highest comment density here and haven't really added anything to the discussion

**Comment by Spacedong** [comment]
this PR is utterly useless and I've added stuff to the discussion

**Comment by ghost** [comment]
add this because there should be a reward for actively doing something you dont nescessarily need to do,
the rewards are also minute and wont effect anything at it's core, and maybe admins will start catbanning more so this will become useful

**Comment by Spacedong** [comment]
well @oprayx73 i fukd ur mom but it wasn't something i necessarily needed to do but I didn't get a reward for it anyway so think about it

**Comment by Pyko1** [comment]
Just confirmed:
Transform sting on human makes him drop tajaran teeths
Lesser form -> Human form regenerates your own teeth.

**Comment by ghost** [comment]
@Pyko1 huh, really? The humanize() proc doesn't seem to have any references to that, I must have missed something. Actually, it's probably just something with the transform sting, since teeth aren't changed when you go monkey

Either way the amount of stuff in vending machines is heavily limited now so it's not a big deal.

**Comment by Pyko1** [comment]
I'm pretty sure teeths are not organs, but are just created by the mop type which dropped em.

**Comment by ghost** [comment]
Teeth are external organs.

**Comment by ghost** [comment]
Oh wait nevermind, they're just in the same file

fml

**Comment by Nokington** [comment]
is this a shitpost PR? Jesus. I don't even want to read all of these comments. The idea of this is good but the prizes are shit.

**Comment by ghost** [comment]
@Nokington No, it's not, that's just spacedong being spacedong. What would you suggest for prizes?

**Comment by Spacedong** [comment]
hey rude

**Comment by Carbonhell** [comment]
I don't like how you coded it. You added a new define and other stuff for a different coin, you could've made it give different prizes related on what kind of coin is being used IE diamond,uranium, steel, cat teeth. The prizes should be something not useful but fun to have, not a fucking claymore/all access/whatever you suggested but something like... i can't think of anything as for now but something that has only an aesthetic change..perhaps ian suit or cosby sweater,unsure.

**Comment by ghost** [comment]
@Carbonhell the prizes are heavily nerfed and their quantities reduced in the lastest commit

I don't know about an Ian suit, those are easy enough to get by ordering corgis, and most clothing shit can be gotten from vendors.

**Comment by Spacedong** [comment]
why don't you just show us what the current selection is instead of saying you nerfed them

**Comment by ghost** [comment]
@spacedong why don't you read the code 

wew lad

**Comment by Carbonhell** [comment]
A pr is supposed to describe what it's adding without having to read the code so uhh

**Comment by Nokington** [comment]
Well first how do they make the coin? Put 32 teeth on a table and use table construction?

**Comment by ghost** [comment]
@spacedong gloves, retro lasers, switchblades, telescopic batons, riot foam darts for the greytide

lasers, advanced tasers, disablers for security after the armory inevitably gets looted by aforementioned greytide/Camille

@Nokington use a welding tool on 32 cat teeth or 48 Xeno teeth.

@Carbonhell I mean I would, but people keep suggesting new stuff so

**Comment by Nokington** [comment]
hey fuck you carbon, im need reading the top of the pr

**Comment by Spacedong** [comment]
what the fuck are you kidding me , retro laser is overpowered, switchblades are ok i guess, telescopic baton is meh, riiot foam darts are the best imo

**Comment by ghost** [comment]
Can't you take the safety off of riot foam darts, which gives them about 25 force or some shit?

Retro lasers aren't that great aside from ranged capabilities, and the average greyshirt can't charge them

**Comment by Spacedong** [comment]
"average greyshirt can't charge them"

except all of them can, they can just weld into any security  checkpoint wall and charge them

**Comment by ghost** [comment]
And a retro laser isn't what I would call OP anyway

**Comment by Spacedong** [comment]
retro laser is OP since anyone can get it by killing a cat ez

**Comment by ghost** [comment]
On lowpop there aren't even cats every round. There might be 3 cats on a 90-player round, if everyone can get rewards for hunting them probability doesn't dictate that you're going to be the one to get it

And there are only 2 retro lasers. You'll get about 30 people hunting cats if this PR gets merged.

**Comment by ghost** [comment]
I probably should have specified, but there are only 2 of every item aside from insulated gloves and darts. Highly unlikely that you'll get them.

**Comment by Pyko1** [comment]
On every vender of that type, currently we have 3 sec venders, and 2 tool venders aboard the station, and i think there is more in space.

**Comment by ghost** [comment]
I forgot to account for the second tool vendor. But you do need engi-access to get to that one, no?

**Comment by Pyko1** [comment]
or a toolbox

**Comment by ghost** [comment]
That's true, but the amount of cats still limits this. And new cats can't be(legally) created outside of a traitor geneticist/changelings round, and even on a Changeling round it's unlikely that they would obtain cat DNA. The chances that a plethora of cats will be created is about the same a wiznerd spawning guns, which unarguably fucks up the round a LOT more.

**Comment by Pyko1** [comment]
Most of the time you can just walk in to the bar and there is at least 1 cat hanged somewhere around, unless chef already butchered it.

**Comment by ghost** [comment]
Maybe 20% of the time. That's not an extremely common sight nowadays. And if a cat is strung up in the bar, it's extremely unlikely that one guy will get all of the teeth.

**Comment by Pyko1** [comment]
Also you don't need to be a traitor geneticist to make cats, they have 8 monkeys or so, and they can always order monkey cubes.

**Comment by ghost** [comment]
@Pyko1 I didn't consider that, but they still need cat DNA. And if they make a fuckton of cats, there are only so many useful things they can get from the vendors. They can get a few of each item, sure, but by that time the greytide probably would have looted most of the machines.

To be honest a lot of these situations you're bringing up are valid points but aren't that likely to happen, and won't affect muh balancings of the average round.

**Comment by Spacedong** [comment]
can we get this PR up to 100 comments

**Comment by ghost** [comment]
@Spacedong probably

**Comment by Spacedong** [comment]
everyone comment on this PR get it up to 100

**Comment by TrashFiend** [comment]
idk this seems overpowered

**Comment by Spacedong** [comment]
yeah it looks overpowered

**Comment by Spacedong** [comment]
get this up to 100 replies

**Comment by Spacedong** [comment]
o ye

**Comment by ghost** [comment]
@dragonrobot Can't you find a lot of random overpowered shit in space

I think that's easier than de-teething one of the few cats on the station before the other 50 assistants do

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -29,12 +30,14 @@
 	var/list/products	= list()	//For each, use the following pattern:
 	var/list/contraband	= list()	//list(/type/path = amount,/type/path2 = amount2)
 	var/list/premium 	= list()	//No specified amount = only one in stock
+	var/list/special = list()
```
Comment (sp1d3rx) [inline code/game/machinery/vending.dm]: please fix the spacing to match existing.

Comment (ghost) [inline code/game/machinery/vending.dm]: okay

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -331,12 +351,22 @@
...
 		else
 			var/list/display_records = product_records
+
 			if(extended_inventory)
 				display_records = product_records + hidden_records
+
+			if (special_coin)
+				display_records = product_records + special_records
+
+			if (special_coin && extended_inventory)
```
Comment (sp1d3rx) [inline code/game/machinery/vending.dm:361]: reverse the order of the if(s). You're overwriting display_records 3x if special coin and extended inventory.

File: code/game/machinery/vending.dm [hunk 4]
```diff
@@ -331,12 +351,22 @@
 			dat += "<font color = 'red'>No product loaded!</font>"
 		else
 			var/list/display_records = product_records
-			if(extended_inventory)
+
```
Comment (ghost) [inline code/game/machinery/vending.dm:354]: The changes below should have a similar effect to what you suggested @sp1d3rx

**Comment by TrashFiend** [comment]
@Cherkir That still requires you to go break into derelict or that white ship with the guns on it or whatnot. That takes a fair amount of time and jetpack fuel if you don't win the z-level lottery. I haven't done a ton of exploring on derelict but isn't it mainly just random guns and some armor, though?

I could more imagine this leading to people fighting over cat corpses and people leaving their duties to hunt down Meow McFluffers the moment they make themselves known.

The first few rounds this is in would probably be a really fun shitfest to watch with people murdering each other over cats, however.

**Comment by ghost** [comment]
@dragonrobot True, but hunting a competent furfag can be pretty difficult and time costly. Giving greytiders a reason to hunt cats would also probably lessen the roundly security/HOP griff.

And yeah, that would be hilarious. Security probably wouldn't mind assistants murdering eachother rather than them for once.(So support this spacedong you cuck)

**Comment by TrashFiend** [comment]
@Cherkir You are giving the greytiders guns, though. The HoP may be spared but the brig will never be the same.

**Comment by ghost** [comment]
@dragonrobot Guns that some of them would get anyway by breaking into EVA and bumming around in space. Punching out a kitty's teeth before hanging them(for good measure) to get the gun just makes it that much more fun

**Comment by Pyko1** [comment]
Note to self: If this gets added and I ever play a cat, remember to punch out a teeth at roundstart and space/hide em, just to see them rage about 1 missing teeth.

**Comment by TrashFiend** [comment]
@Cherkir Your average assistant really isn't going to manage to break into the armory and not get bent by security or admins if they decide to fight back. Breaking into the armory and stealing guns is some 100% traitorous "you're getting perma'd and never getting out if we catch you" shit. They can't really use anything but tasers/disablers against security when they try to run or they are literally non-antags murdering implanted officers for no raisins.

The whole problem of assistants possibly breaking into the armory kind of corrects itself.

**Comment by ghost** [comment]
@dragonrobot Aside from that, this might make assistants not want to break into the armory which means the HOS/Captain/HOP doesn't have to be Camille Driggers and hide the entire armory within the first five minutes of every shift.

**Comment by ghost** [comment]
any more input for this PR?

**Comment by danhibiki** [comment]
> 113 comments.

I am not reading every single one of these, so I'll put it simple. You need to condense what the "prizes" are for these coins in the original post and we can have, hopefully, a structured discussion about this PR

**Comment by ArcLumin** [comment]
Why not merge this with mining? As in, we already have a mining rewards vendor, just make cat coins give mining points, add in more mining rewards (like the ones above) and allow an all access rewards vendor. Rather than 32 teeth for one coin, 8 teeth for 1 coin, and 1 coin equals, say 300~500 points. It's a needed boost to mining as well.

**Comment by TrashFiend** [comment]
At some point you might as well go almost full goon and call them credits, and let people buy stuff with credits.

**Comment by ArcLumin** [comment]
I'd be perfectly fine with that. Coins are almost useless in game right now, only for a select few vendors to get stuff like cigars or insuls. Making a sort of currency would be nice overall

**Comment by ghost** [comment]
Sorry about my inactivity, the current prizes are in the main post if anyone missed them.

And yeah, a full overhaul with coins would be nice, but that would require a new PR.

**Comment by ArcLumin** [comment]
Well, I don't think it's super great in it's current form, but a full currency overhaul would be preferable. Perhaps scrap this pull and make a new one with a full overhaul?

**Comment by TDcoolguy300** [comment]
The uprising of NPC merchants is upon us.

**Comment by ghost** [comment]
That could be done. What do heads think about this PR anyway if I'm going to close it?

**Comment by Nokington** [comment]
Come up with better prizes that aren't antag items, all access cards and guns and then make a new PR with it.

**Comment by ghost** [comment]
None of the prizes are antag items or all access cards, and the only gun is shit.

Have you looked at the prizes recently? And why would I make a new PR?

**Comment by Nokington** [comment]
No because I am not reading through all that again.

**Comment by ghost** [comment]
The prizes are in the first comment

**Comment by nalarmg** [comment]
Heads, consensus on merging this?

**Comment by danhibiki** [comment]
still -1 with retrolasers in vending machines from me

**Comment by Nokington** [comment]
Yeah I still don't think having any weapons at all is a good idea.

**Comment by nalarmg** [comment]
Two against, no for that I can reasonably tell. No visible efforts at comforming to head admin feedback. Closing for now. Feel free to reopen this or submit a new PR if there's further interest.

**Pull request closed without merging**